README.md

```diff
@@ -4,43 +4,98 @@ This repository contains data to test, develop and debug
-| `Be_snapshot0-3.in.npy`                   | Bispectrum descriptors numpy array.                                                                                          |
+File Name                       | Description
+-|-
+`recreate_data/`                | Input scripts for QE
+`cubes/`                        | `.cube` files for the local density of states
+`Be.pbe-n-rrkjus_psl.1.0.0.UPF` | Pseudopotential used for the QE calculation
+`Be_snapshot0.dens.npy`         | Electronic density numpy array (snapshot 0)
+`Be_snapshot.dens.h5`           | Electronic density (HDF5 format, see details below)
+`Be_snapshot0.dos.npy`          | Density of states numpy array (snapshot 0)
+`Be_snapshot0-3.out`            | Output file of QE. calculation
+`Be_snapshot0-3.in.npy`         | Bispectrum descriptors numpy array
+`Be_snapshot0-3.out.npy`        | Local density of states numpy array
+`Be_snapshot0-3.in.h5`          | Bispectrum descriptors (HDF5 format)
+`Be_snapshot0-3.out.h5`         | Local density of states (HDF5 format)
+
+### `numpy` format files
+
+SNAP bispectrum descriptors of length 91 on `18 x 18 x 27` real space grid.
+
+> [!NOTE]
+> In the last dimension of length 94, the first 3 entries are the grid coordinates / indices (an artifact of the SNAP vector generation). **The actual features are `snap_array[..., 3:]`**.
 
-The `.npy` arrays have the following shapes:
 
 ```py
->>> np.load('Be2/Be_snapshot0.dos.npy').shape
-(11,)
-´>>> np.load('Be2/Be_snapshot0.dens.npy').shape
-(18, 18, 27, 1)
 >>> np.load('Be2/Be_snapshot1.in.npy').shape
 (18, 18, 27, 94)
+```
+
+LDOS (11 points) on `18 x 18 x 27` real space grid.
+
+```py
 >>> np.load('Be2/Be_snapshot1.out.npy').shape
 (18, 18, 27, 11)
 ```
 
+Density of states (only provideed for snapshot 0):
+
+```py
+>>> np.load('Be2/Be_snapshot0.dos.npy').shape
+(11,)
+```
+
+Density for snapshot 0 on a `18 x 18 x 27` real space grid. The extra dimension
+can be ignored, i.e. use `d=np.load(...); d[..., -1]` to squeeze the shape to
+`(18, 18, 27)`.
+
+```py
+>>> np.load('Be2/Be_snapshot0.dens.npy').shape
+(18, 18, 27, 1)
+```
+
+
+### OpenPMD-based files
+
+MALA [supports the OpenPMD
+format](https://mala-project.github.io/mala/advanced_usage/openpmd.html), so we
+also provide data in that format here.
+
+```sh
+$ h5ls -r Be_snapshot0.in.h5 | grep Dataset | sort -V
+/data/0/meshes/Bispectrum/0  Dataset {18, 18, 27}
+/data/0/meshes/Bispectrum/1  Dataset {18, 18, 27}
+...
+/data/0/meshes/Bispectrum/93 Dataset {18, 18, 27}
+
+$ h5ls -r Be_snapshot0.out.h5 | grep Dataset | sort -V
+/data/0/meshes/LDOS/0     Dataset {18, 18, 27}
+/data/0/meshes/LDOS/1     Dataset {18, 18, 27}
+...
+/data/0/meshes/LDOS/10    Dataset {18, 18, 27}
+```
+
+For the density, the snapshot number 0 is encoded in the dataset name.
+
+```sh
+$ h5ls -r Be_snapshot.dens.h5 | grep Dataset
+/data/0/meshes/Density/0 Dataset {18, 18, 27}
```

Comment:
@franzpoeschel could you please help out here and specify which 0 (I assume the second, `.../Density/0`) specifies the snapshot number? I think this is not documented yet in https://mala-project.github.io/mala/advanced_usage/openpmd.html. Also, I;m not sure what `/data/0/` signifies. Maybe we could document that as well.

Reply:
I'll have a look.

Reply:
Concerning the documentation, the most precise term to call this would be the`openPMD iteration`. Maybe additionally using the output of `openpmd-ls` could help make this more precise:

```
> openpmd-ls Be_snapshot.dens.h5 
openPMD series: Be_snapshot.dens
openPMD standard: 1.1.0
openPMD extensions: 0

data author: ...
data created: 2023-05-23 15:37:18 +0200
data backend: HDF5
generating machine: unknown
generating software: MALA (version: 1.1.0)
generating software dependencies: unknown

number of iterations: 1 (groupBased)
  all iterations: 0 

number of meshes: 1
  all meshes:
    Density

number of particle species: 0
```

If you want to further specify, you might also add something like `Iterations in openPMD are encoded either as a part of the filename (file-based iteration encoding) or, if multiple iterations are stored in a single file, as a hierarchical group inside the file, e.g. `/data/0`.

Btw:
Good that you raised this, MALA currently actually *does* hardcode the snapshot in the filename instead of leaving this up to openPMD. The consequence is that it's not possible to open the entire Series within the same openPMD context. I'll prepare a fix.

Reply:
Thanks a lot. 

So multiple snapshots (we indeed usually come from an MD run) would be modeled as iterations in a Series, such as 

```
/data/0/meshes/Density/0
/data/1/meshes/Density/0
/data/2/meshes/Density/0
...
```

Reply:
Resolved by 94c61dd2329c8788dfb8b4cd5b82860ea61c3a95